src/main/java/com/money_account_service/controllers/AccountController.java

```diff
@@ -1,51 +1,33 @@
 package com.money_account_service.controllers;
 
 import com.money_account_service.dtos.request.CreateAccountRequestDto;
-import com.money_account_service.dtos.response.AuthorizeResponseDto;
 import com.money_account_service.mappers.UserMapper;
-import com.money_account_service.models.UserModel;
 import com.money_account_service.services.AccountService;
-import com.money_account_service.utility.Authorize;
-import com.money_account_service.utility.AuthorizeRequestInterceptor;
+import com.money_account_service.utility.UserServiceClient;
 import jakarta.servlet.http.HttpServletRequest;
 import lombok.AllArgsConstructor;
-import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
-import org.springframework.web.server.ResponseStatusException;
-
-import java.util.Optional;
 
 @RestController
 @AllArgsConstructor
 public class AccountController {
 
     private final AccountService accountService;
 
-    private final AuthorizeRequestInterceptor authorizeRequestInterceptor;
+    private final UserServiceClient userServiceClient;
 
     @PostMapping("/account")
-    public void createAccount(@RequestBody CreateAccountRequestDto createAccountRequestDto, @Authorize UserModel userModel) {
-        System.out.println(authorizeResponseDto);
-//        if (authorizeResponseDto.isPresent()) {
-//            UserModel userModel = UserMapper.authorizeRequestDtoToUserModel(authorizeResponseDto.get());
-//            accountService.createAccount(createAccountRequestDto, userModel);
-//        } else {
-//            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Failed to authorize user");
-//        }
+    public void createAccount(@RequestBody CreateAccountRequestDto createAccountRequestDto, HttpServletRequest request) {
+        accountService.createAccount(createAccountRequestDto,
+                UserMapper.authorizeRequestDtoToUserModel(userServiceClient.authorize(request.getHeader("Authorization"))));
```

Comment:
Ok, we can do it programmatically, which also has a big advantage - **readability**.

I would suggest the following imporvment:
1. Create a base class for controllers which need to authenticated, e.g. `abstract class AuthController`.
2. All controller will inherit from it.
3. The AuthController has `AuthorizationService`, which has one method `authorize: HttpServletRequest -> User`. The method retruns the user or throws `Forbidden` exception.
4. Then the `UserServiceClient` and request mapping is hidden. It will be helpful in all other endpoints you will write. We'll save quite a few lines of code.
5. A small improvment could also be to bind a specific header by name, insteal of the whole servlet request https://www.baeldung.com/spring-rest-http-headers